Lua error when closing cap_viewer menu

I seem to be getting an error when closing the cap_viewer. 

This is the error:

[ERROR] addons/gcap/lua/autorun/client/gcap_client.lua:195: Tried to use invalid object (type Panel) (Object was NULL or not of the right type)
1. SetHTML - [C]:-1
   1. func - addons/gcap/lua/autorun/client/gcap_client.lua:195
   2. unknown - lua/includes/modules/net.lua:32

--- a/gcap/lua/autorun/client/gcap_client.lua
+++ b/gcap/lua/autorun/client/gcap_client.lua
@@ -179,7 +179,8 @@ concommand.Add("cap_viewer", function(ply, cmd, args)
             end
         end)
 
-        local htmlpnl
+        local htmlpnl = frm:Add( "HTML" )
+        htmlpnl:Dock( FILL )
 
         net.Receive("addPicture", function(len, server)
             ply = LocalPlayer()
@@ -195,10 +196,7 @@ concommand.Add("cap_viewer", function(ply, cmd, args)
                 htmlpnl:SetHTML( '<style type="text/css"> body { margin: 0; padding: 0; overflow: hidden; } img { width: 100%; height: 100%; } </style> <img src="data:image/jpg;base64,' .. data .. '"> ')
                 ply.ScreenViewChunks = {}
             end
-        end)       
-
-        htmlpnl = frm:Add( "HTML" )
-        htmlpnl:Dock( FILL )
+        end)
     end
 end)
 

